pexpect/fdpexpect.py

```diff
@@ -112,3 +113,19 @@ def writelines(self, sequence):
         "Call self.write() for each item in sequence"
         for s in sequence:
             self.write(s)
+
+    def read_nonblocking(self, size=1, timeout=-1):
+        '''The read_nonblocking method of fdspawn assumes that the file
+        will never block. This is not the case for all file like objects
```

Comment:
It's better, but ideally the docstring should start by describing what this method does, rather than the method on the parent class. I like docstrings to follow the format:

```
One line overview of functionality
<blank line>
If necessary, more detailed description of options, return value, and rationale.
```